ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/imageCveSingle.test.js

```diff
@@ -0,0 +1,173 @@
+import { visitWorkloadCveOverview } from './WorkloadCves.helpers';
+import { selectors } from './WorkloadCves.selectors';
+
+describe('Image CVE Single page', () => {
+    withAuth();
+
+    it('should correctly handle ImageCVE single page specific behavior', () => {
+        // Apply global default filters
+        visitWorkloadCveOverview();
+
+        // Click any CVE link in the table to visit the ImageCVE Page.
+        cy.get('tbody tr td a:contains("CVE")').first().click();
+
+        // Wait for the summary card skeleton to disappear
+        cy.get('.pf-c-skeleton:contains("Loading image cve summary data")').should('not.exist');
+
+        // Check that only applicable resource menu items are present in the toolbar
+        cy.get(selectors.resourceDropdown).click();
+        cy.get(selectors.resourceMenuItem('CVE')).should('not.exist');
+        cy.get(selectors.resourceMenuItem('Image'));
+        cy.get(selectors.resourceMenuItem('Deployment'));
+        cy.get(selectors.resourceMenuItem('Cluster'));
+        cy.get(selectors.resourceMenuItem('Namespace'));
+        cy.get(selectors.resourceDropdown).click();
+
+        // The Images toggle button should be the default and the count of the images in the button
+        // should match the count of the images in the "Affected images" card
+        cy.get('button[aria-pressed="true"]:contains("Image")').then((button) => {
+            const buttonImageCount = button.text().replace(/ Image.*/, '');
+
+            cy.get(
+                `${selectors.summaryCard('Affected images')} *:contains("images affected")`
+            ).then((affectedImages) => {
+                const affectedImageCount = affectedImages
+                    .text()
+                    .replace(/\/\d* images affected/, '');
+
+                expect(buttonImageCount).to.equal(affectedImageCount);
+            });
+        });
+    });
+
+    it('should correctly handle local filters on the images tab', () => {
+        visitWorkloadCveOverview();
+        cy.get('tbody tr td a:contains("CVE")').first().click();
+
+        cy.get('table tbody tr:nth-of-type(1) td[data-label="CVE severity"]').then(
+            ([$severity]) => {
+                // Extract the severity from the first row of the table. These are values
+                // that we know should exist in the table when no filters are applied.
+                const severity = $severity.innerText;
+
+                // Apply any filter that _doesn't_ match the severity value extracted from the first row to ensure all data
+                // matching that severity is filtered out from the table. e.g. If the first row has a severity of "Critical",
+                // applying a filter of "Important" will ensure that no data in the table will match a severity of "Critical".
+                cy.get(selectors.severityDropdown).click();
+                cy.get(`${selectors.severityMenuItems} label:not(:contains("${severity}"))`)
```

Comment:
Oh, nice! I have not thought about `:not` pardon pun.

Reply:
And the example below of double negative is worth to share at a team meeting.

Reply:
Sure, it is a bit odd to express "select an item from this dropdown that _isn't_ this value".

I don't like the non-determinism of it, e.g. we don't really know what value we are going to get, but it seemed like a good approach to be able to test the filtering of the data without knowing what the data may contain at any given time.

If we are able to add a standard suite of e2e test data to the environment these types of problems should mostly go away though.